### PR TITLE
Add eslint config

### PR DESCRIPTION
**Pull request by jacobtylerwalls**
## Overview
We were (?) getting some kind of minimal eslint from craco that had the potential to fail a PR for some basic things, but not anything more than that, and no linting of test files, and no integration with pre-commit. Unfortunately, MegaLinter doesn't recognize `react-app` as a preset, so the config got a little bloated.

Also, forward-port the MegaLinter fix from azavea/azavea-project-template#81

### Notes
Tested this against the app template and included items necessary there (e.g. adding `cypress` as a permitted global).

## Testing Instructions

- Edit some TS(x)
- `scripts/lint` or stage your changes and `scripts/lint staged`
  - [x] Ensure lint script finds lint
  - [x] Ensure lint script exits cleanly, doesn't hang if interrupted once before
 
 ## Checklist

- [ ] `fixup!` commits have been squashed
- [ ] `CHANGELOG.md` updated with summary of features or fixes, following [Keep a Changelog](https://keepachangelog.com/en/1.0.0/) guidelines
- [ ] `README.md` updated if necessary to reflect the changes
- [ ] CI passes after rebase

### CODE REVIEW

File: src/app/src/cachedApiQuery.ts [hunk 1]
```diff
@@ -18,7 +18,7 @@ import transportationSpending from './data/Transportation.spending.json';
 import allSpendingOverTime from './data/monthly.spending.json';
 
 const cachedApiQuery: typeof fetchBaseQuery = _ => {
-    return (stringOrArgs, api) => {
+    return (stringOrArgs, _api) => {
```
Comment (mstone121) [inline src/app/src/cachedApiQuery.ts]: Maybe we could just remove this arg if it isn't used.

File: src/app/package.json [hunk 2]
```diff
@@ -52,7 +52,36 @@
     "fetch-data:agency": "ts-node dataScripts/fetchAgencyData.ts"
   },
   "eslintConfig": {
-    "extends": "react-app"
+    "env": {
```
Comment (mstone121) [inline src/app/package.json]: Any way we could move this into an `.eslintrc`?

**Review by mstone121 — APPROVED** [review]
Working well. Nice to have linting again!

**Comment by jacobtylerwalls** [comment]
@mstone121 What do you think about these:
```
green-equity-demo-app-1  | WARNING in [eslint] 
green-equity-demo-app-1  | src/cachedApiQuery.ts
green-equity-demo-app-1  |   Line 69:17:  Forbidden non-null assertion  @typescript-eslint/no-non-null-assertion
```

**Comment by mstone121** [comment]
Generally I'm not a fan of using non-null assertions, but there are cases like this one where it is pretty clear that the variable in question is not null (based on earier logic, see line 65).

Something like this:
```js
    if (request.geo_layer_filters) { // <-- line 65
        return {
            ...spending,
            results: spending.results.filter(
                result =>
                    request.geo_layer_filters &&
                    request.geo_layer_filters.includes(result.shape_code)
            ),
        };
    }
```
removes the error, but adds an unecessary boolean check to each iteration of the filter function.

In this case, I think it's okay to have non-null assertion and hopefully one day typescript will become smart enough to see that the value is not null. (It is already smart enough to discriminate unions types, and essentially all this is is a union type between `string[]` and `undefined`, so I'm not sure why it doesn't discriminate based on line 65.)

**Comment by jacobtylerwalls** [comment]
I realized that the no non-null assertion rule was a warning, so it wasn't blocking CI. We can always address it separately or silence it.

**Pull request merged**